bugsnag-spring/src/test/java/com/bugsnag/SpringMvcTest.java

```diff
@@ -0,0 +1,291 @@
+        // the current session count before the test starts instead.
+        sessionsStartedBeforeTest = getSessionCount();
+    }
+
+    @Test
+    public void bugsnagNotifyWhenUncaughtControllerException() {
+        callRuntimeExceptionEndpoint();
+
+        Report report = verifyAndGetReport();
+
+        // Assert that the exception was detected correctly
+        assertEquals("Test", report.getExceptionMessage());
+        assertEquals("java.lang.RuntimeException", report.getExceptionName());
+
+        // Assert that the severity, severity reason and unhandled values are correct
+        assertEquals(Severity.ERROR.getValue(), report.getSeverity());
+        assertEquals(
+                SeverityReasonType.REASON_UNHANDLED_EXCEPTION_MIDDLEWARE.toString(),
+                report.getSeverityReason().getType());
+        assertThat(
+                report.getSeverityReason().getAttributes(),
+                is(Collections.singletonMap("framework", "Spring")));
+        assertTrue(report.getUnhandled());
+    }
+
+    @Test
+    public void noBugsnagNotifyWhenSendUncaughtExceptionsFalse() {
+        bugsnag.getConfig().setSendUncaughtExceptions(false);
+
+        callRuntimeExceptionEndpoint();
+
+        verifyNoReport();
+    }
+
+    @Test
+    public void bugsnagSessionStartedWhenAutoCaptureSessions() {
+        callRuntimeExceptionEndpoint();
+
+        assertSessionsStarted(1);
+    }
+
+    @Test
+    public void noBugsnagSessionStartedWhenAutoCaptureSessionsFalse() {
+        bugsnag.getConfig().setAutoCaptureSessions(false);
+
+        callRuntimeExceptionEndpoint();
+
+        assertSessionsStarted(0);
+    }
+
+    @Test
+    public void requestMetadataSetCorrectly() {
+        callRuntimeExceptionEndpoint();
+
+        Report report = verifyAndGetReport();
+
+        // Check that the context is set to the HTTP method and URI of the endpoint
+        assertEquals("GET /throw-runtime-exception", report.getContext());
+
+        // Check that the request metadata is set as expected
+        @SuppressWarnings(value = "unchecked") Map<String, Object> requestMetadata =
+                (Map<String, Object>) report.getMetaData().get("request");
+        assertEquals("http://localhost:" + randomServerPort + "/throw-runtime-exception",
+                requestMetadata.get("url"));
+        assertEquals("GET", requestMetadata.get("method"));
+        assertEquals("127.0.0.1", requestMetadata.get("clientIp"));
+
+        // Assert that the request params are as expected
+        @SuppressWarnings(value = "unchecked") Map<String, String[]> params =
+                (Map<String, String[]>) requestMetadata.get("params");
+        assertEquals("paramVal1", params.get("param1")[0]);
+        assertEquals("paramVal2", params.get("param2")[0]);
+
+        // Assert that the request headers are as expected
+        @SuppressWarnings(value = "unchecked") Map<String, String> headers =
```

Comment:
Are we assuming that headers only have a single value? If I recall correctly, the HTTP standard allows multiple values to be set as values for headers.

Reply:
I've added support for multiple headers by concatenating multiple values to a comma separated string.